.github/copilot-instructions.md

```diff
@@ -0,0 +1,334 @@
+
+#### `nvm unload`
+- Completely removes nvm from shell environment
+- Unsets all nvm functions and variables
+
+### Internal Function Categories
+
+#### Version Resolution
+- `nvm_resolve_alias()` - Resolves aliases to version numbers
+- `nvm_version()` - Finds best matching local version
+- `nvm_remote_version()` - Finds best matching remote version
+- `nvm_normalize_version()` - Standardizes version strings
+
+#### Installation Helpers
+- `nvm_install_binary()` - Downloads and installs precompiled binaries
+- `nvm_install_source()` - Compiles Node.js from source
+- `nvm_download_artifact()` - Downloads tarballs or binaries
+- `nvm_compute_checksum()` - Verifies download integrity
+
+#### Path Management
+- `nvm_change_path()` - Updates PATH for version switching
+- `nvm_strip_path()` - Removes nvm paths from PATH
+- `nvm_version_path()` - Gets installation path for version
+
+#### Utility Functions
+- `nvm_echo()`, `nvm_err()` - Output functions
+- `nvm_has()` - Checks if command exists
+- `nvm_is_zsh()` - Shell detection
+- `nvm_sanitize_path()` - Cleans sensitive data from paths
+
+## Running Tests
+
+### Test Framework
+nvm uses the **urchin** test framework for shell script testing.
+
+### Test Structure
+```
+test/
+├── fast/           # Quick unit tests
+├── slow/           # Integration tests
+├── sourcing/       # Shell sourcing tests
+├── install_script/ # Installation script tests
+├── installation_node/ # Node installation tests
+└── common.sh       # Shared test utilities
+```
+
+### Running Tests
+
+#### Install Dependencies
+```bash
+npm install  # Installs urchin, semver, and replace tools
+```
+
+#### Run All Tests
+```bash
+npm test               # Runs tests in current shell
+make test              # Runs tests in all supported shells
+make test-bash         # Runs tests only in bash
+make test-zsh          # Runs tests only in zsh
+```
+
+#### Run Specific Test Suites
+```bash
+make TEST_SUITE=fast test        # Only fast tests
+make TEST_SUITE=slow test        # Only slow tests
+make SHELLS=bash test            # Only bash shell
+```
+
+#### Individual Test Execution
+```bash
+./test/fast/Unit\ tests/nvm_get_arch     # Run single test
+urchin test/fast/                        # Run fast test suite
+```
+
+### Test Writing Guidelines
+- Tests should work across all supported shells (sh, bash, dash, zsh)
+- Use `die()` function for test failures
+- Clean up after tests in cleanup functions
+- Mock external dependencies when needed
+- Place mocks in `test/mocks/` directory
+
+## Shell Environment Setup
+
+### Supported Shells
+- **bash** - Full feature support
+- **zsh** - Full feature support  
+- **dash** - Basic POSIX support
+- **sh** - Basic POSIX support
+- **ksh** - Limited support (experimental)
+
+### Installing Shell Environments
+
+#### Ubuntu/Debian
+```bash
+sudo apt-get update
+sudo apt-get install bash zsh dash ksh
+```
+
+#### macOS
+```bash
+# zsh is default, install others via Homebrew
```

Comment:
bash is available on macs by default, it's just not the default shell of a new user account